### ISSUE TITLE
Crash when closing a scene with a visible window

### ISSUE THREAD

**Issue by Rindbee** [godotengine/godot#65146]
### Godot version

v4.0.alpha15.official [432b25d36], 8fa9d1ae4

### System information

Linux  mint 20.3, NVIDIA GeForce GTX 1050 Ti

### Issue description

Closing a scene containing a window, when the window is visible, crashes; when the window is not visible, everything works fine.
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.alpha.custom_build (8fa9d1ae467632daa5f12b278eb616fd580a94ec)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x43090) [0x7f6dcc495090] (??:0)
[2] /home/bee/code/godot/godot/bin/godot.linuxbsd.opt.tools.x86_64.llvm() [0x687e7f6] (??:?)
[3] /home/bee/code/godot/godot/bin/godot.linuxbsd.opt.tools.x86_64.llvm() [0x3c47a04] (/home/bee/code/godot/godot/./core/object/object.h:737)
[4] /home/bee/code/godot/godot/bin/godot.linuxbsd.opt.tools.x86_64.llvm() [0x65a8a86] (/home/bee/code/godot/godot/core/object/message_queue.cpp:230)
[5] /home/bee/code/godot/godot/bin/godot.linuxbsd.opt.tools.x86_64.llvm() [0x65a8d59] (/home/bee/code/godot/godot/core/object/message_queue.cpp:?)
[6] /home/bee/code/godot/godot/bin/godot.linuxbsd.opt.tools.x86_64.llvm() [0x47651ae] (/home/bee/code/godot/godot/scene/main/scene_tree.cpp:?)
[7] /home/bee/code/godot/godot/bin/godot.linuxbsd.opt.tools.x86_64.llvm() [0x269dcde] (/home/bee/code/godot/godot/main/main.cpp:2968)
[8] /home/bee/code/godot/godot/bin/godot.linuxbsd.opt.tools.x86_64.llvm() [0x2650b81] (/home/bee/code/godot/godot/platform/linuxbsd/os_linuxbsd.cpp:538)
[9] /home/bee/code/godot/godot/bin/godot.linuxbsd.opt.tools.x86_64.llvm() [0x264b02b] (/home/bee/code/godot/godot/platform/linuxbsd/godot_linuxbsd.cpp:74)
[10] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7f6dcc476083] (??:0)
[11] /home/bee/code/godot/godot/bin/godot.linuxbsd.opt.tools.x86_64.llvm() [0x264aeae] (??:?)
-- END OF BACKTRACE --
================================================================
```

### Steps to reproduce

1. Create a scene;
2. Add a window;
3. Close the scene with the window visible, crash;
4. Close the scene with the window invisible, no error.

**Comment by Rindbee** [comment]
Occasionally it doesn't crash, but gets the following error message:
````
ERROR: Error calling deferred method: 'SceneTreeEditor::SceneTreeEditor::_warning_changed': Cannot convert argument 1 from Object to Object..
    at: _call_function (core/object/message_queue.cpp:231)
````

**Comment by Rindbee** [comment]
This is a regression introduced by #53299.

**Comment by Rindbee** [comment]
I think it has been fixed by #65630.
